markdown_it/tree.py

```diff
@@ -0,0 +1,211 @@
+"""A tree representation of a linear markdown-it token stream.
+
+This module is not part of upstream JavaScript markdown-it.
+"""
+from typing import NamedTuple, Sequence, Tuple, Dict, List, Optional, Any
+
+from .token import Token
+from .utils import _removesuffix
+
+
+class SyntaxTreeNode:
+    """A Markdown syntax tree node.
```

Comment:
perhaps also add a note here, that it should be initialised by `SyntaxTreeNode.from_tokens`